ACMILabs/interactive-label
# #2049 Cache playlist and images

Resolves ACMILabs/xos#2049

- Interactive label now caches playlist json and images
- If no network, it tries to get cached data
- if no cached data and no network, show an error screen with troubleshooting text

### Acceptance Criteria
- [x] When label is booted and can't access XOS within 5 secs then use cached config and content.
- [x] Update https://kb.acmi.net.au/display/DD/XOS+Gallery+Device+Status

### Relevant design files
* None

### Testing instructions
Add a list of steps required to test the feature.
1. Make sure you're connected to the VPN and a playlist is set on the `config.env` such as `XOS_PLAYLIST_ID=46`
2. Run the interactive label locally `cd development` and `docker-compose up --build`
3. Check that the label is displaying as usual: http://0.0.0.0:8081/
4. Disconnect from the VPN
5. Spin up the interactive label again `docker-compose up --build`
6. Check that the previous label is still being displayed http://0.0.0.0:8081/
7. Delete the cache at `data/*`
8. Spin up the interactive label again `docker-compose up --build`
9. Check that a nicely formatted error page is displayed

### DoD
For requester to complete:
~Changelog has been updated if necessary~
~Deployment / migration instruction have been updated if required~
